src/main/groovy/life/qbic/samplestatus/reporter/Result.groovy

```diff
@@ -0,0 +1,137 @@
+package life.qbic.samplestatus.reporter
+
+import java.util.function.Function
+
+/**
+ * <b>Class Result</b>
+ *
+ * <p>This class introduces the Rust idiom to use Result objects to enforce the client code
+ * to apply some best practises to safe value access and proper exception handling.</p>
+ *
+ * <p>Results can be used to wrap an actual value of type <code>V</code> or an exception of type
+ * <code>E</code>. An object of type Result can only contain either a value or an exception, not both in the
+ * same instance.</p>
+ *
+ * To properly deal with Result objects, a good idiom looks like this:
+ *
+ * <pre>
+ *
+ * final Result\\<String, Exception\> result = new Result("Contains actual information")
+ * // access the value
+ * switch (result) {
+ *    case { it.isOk() }:
+ *      doSomething(it.getValue())
+ *      break
+ *    case { it.isError() }:
+ *      doErrorHandling(it.getError())
+ *      break
+ * }
```

Comment:
Using the enhanced switch statements introduced in java 14, this would look like the following:
```
 * Function action = switch (result) {
 *   case result.isOk() : Function<V, ?>
 *   case result.isError() : Function<E, ?>
 * }
 * action.apply(result)
```
```
 * Consumer consumer = switch (result) {
 *   case result.isOk() : Consumer<V>
 *   case result.isError() : Consumer<E>
 * }
 * action.accept(result)
```
```
 * switch (result) {
 *   case result.isOk() : () -> {}
 *   case result.isError() : () -> {}
 * }
```

Reply:
uhhhh i like! Thanks a lot